src/App.jsx

```diff
@@ -1,13 +1,187 @@
-
 import './App.css'
+import {useState} from "react";
+import Fruit from "./components/Fruit.jsx";
+import InputField from "./components/InputField.jsx";
+import Button from "./components/Button.jsx";
+import TextField from "./components/TextField.jsx";
 
 function App() {
 
-  return (
-    <>
-        <h1>Fruitmand bezorgservice</h1>
-    </>
-  )
+    const [fruitAmount, setFruitAmount] = useState({
+        aardbeien: 0,
+        bananen: 0,
+        appels: 0,
+        kiwis: 0
+    });
+
+    const [formState, setFormState] = useState({
+        firstName: '',
+        lastName: '',
+        age: 0,
+        zipcode: '',
+        freq: 'weekly',
+        dayTime: '',
+        remark: '',
+        agreement: false,
+    });
+
+    function handleSubmit(e) {
+        e.preventDefault();
+        console.log(fruitAmount);
+        console.log(formState);
+    }
+
+    function handleFruitClick(e) {
+        const changedFieldName = e.currentTarget.name;
+        const newValue = fruitAmount[changedFieldName] +
+            ((e.currentTarget.id === "add")? 1 : -1);
+        setFruitAmount({
+            ...fruitAmount,
+            [changedFieldName]: newValue,
+        });
+    }
+
+    function handleFormChange(e) {
+        const changedFieldName = e.target.name;
+        const newValue = e.target.type === "checkbox" ? e.target.checked : e.target.value;
+
+        setFormState({
+            ...formState,
+            [changedFieldName]: newValue,
+
+        });
+    }
+
+    return (
+        <>
+            <h1>Fruitmand bezorgservice</h1>
+            <Fruit
+                title="Aardbeien"
+                id="aardbeien"
+                fruitAmount={fruitAmount.aardbeien}
+                disabled={fruitAmount.aardbeien === 0}
```

Comment:
Dit had je ook bínnen het Fruit component kunnen doen, omdat de counter al meegegeven wordt en de expressie voor elke fruitsoort hetzelfde is. Deze property is daarmee dus overbodig 😉